StRoot/StFcsWaveformFitMaker/StFcsWaveformFitMaker.cxx

```diff
@@ -86,116 +87,277 @@ ClassImp(StFcsWaveformFitMaker)
+      mH1F_NPeaksFiltered[i] = 0;
+      mH1F_Res0[i] = 0;
+      mH1F_Res0Zoom[i] = 0;
+      mH1F_Sum8Res0[i] = 0;
+      mH1F_Sum8Res0Zoom[i] = 0;
+      mH1F_FitRes0[i] = 0;
+      mH1F_FitRes0Zoom[i] = 0;
+      mH2F_Sum8vFit[i] = 0;
+    }
+}
+
+StFcsWaveformFitMaker::~StFcsWaveformFitMaker() {
+  mChWaveData.Delete();//Completely clear all graphs (Put in Finish??)
+  if( mPulseFit!=0 ){delete mPulseFit;}
+  for( UShort_t i=0; i<7; ++i ){
+    if( i<3 ){
+      if( mH2_Dep0DepMod[i]!=0 ){delete mH2_Dep0DepMod[i];}
+      if( mH2_Sum8Dep0[i]!=0   ){delete mH2_Sum8Dep0[i];}
+      if( mH2_Sum8DepMod[i]!=0 ){delete mH2_Sum8DepMod[i];}
+    }
+    if( i<6 ){
+      if( mH2F_AdcTbAkio[i]!=0 )     {delete mH2F_AdcTbAkio[i];}
+      if( mH2F_AdcTbMine[i]!=0 )     {delete mH2F_AdcTbMine[i];}
+      if( mH2F_SumFitvSumWin[i]!=0 ) {delete mH2F_SumFitvSumWin[i];}
+      if( mH2F_APeakvMPeak[i]!=0 )   {delete mH2F_APeakvMPeak[i];}
+      if( mH1F_PeakStart[i]!=0 )     {delete mH1F_PeakStart[i];}
+      if( mH1F_PeakEnd[i]!=0 )       {delete mH1F_PeakEnd[i];}
+      if( mH2F_NOvsId[i]!=0    )     {delete mH2F_NOvsId[i];}
+    }
+    if( mH2F_AdcTbValidPeak[i]!=0 ){delete mH2F_AdcTbValidPeak[i];}
+    if( mH1F_NPeaks[i]!=0 )        {delete mH1F_NPeaks[i];}
+    if( mH1F_NPeaksFiltered[i]!=0 ){delete mH1F_NPeaksFiltered[i];}
+    
+    if( mH1F_Res0[i]!=0      )     {delete mH1F_Res0[i];}
+    if( mH1F_Res0Zoom[i]!=0  )     {delete mH1F_Res0Zoom[i];}
+    if( mH1F_Sum8Res0[i]!=0  )     {delete mH1F_Sum8Res0[i];}
+    if( mH1F_Sum8Res0Zoom[i]!=0 )  {delete mH1F_Sum8Res0Zoom[i];}
+    if( mH1F_FitRes0[i]!=0      )  {delete mH1F_FitRes0[i];}
+    if( mH1F_FitRes0Zoom[i]!=0  )  {delete mH1F_FitRes0Zoom[i];}
+    if( mH2F_Sum8vFit[i]!=0     )  {delete mH2F_Sum8vFit[i];}
+  }
+  if( mTime!=0 )                  { delete mTime; }
+
+  if( mH1_NPeaksAkio!=0 )         {delete mH1_NPeaksAkio;}
+  if( mH1_NPeaksFilteredAkio!=0 ) {delete mH1_NPeaksFilteredAkio;}
+  
+  if( mH1_PeakTiming!=0 )         {delete mH1_PeakTiming;}
+
+  if( mH2_NPeakvsPeakXdiff )      {delete mH2_NPeakvsPeakXdiff;}
+  if( mH2_NPeakvsPeakYratio)      {delete mH2_NPeakvsPeakYratio;}
+  if( mH1_VOverlap!=0 )           {delete mH1_VOverlap;}
+  if( mH2_NOvsNPeaks!=0 )         {delete mH2_NOvsNPeaks;}
+  if( mH2_VvsNOverlap!=0    )     {delete mH2_VvsNOverlap;}
+
+  if( mH1_TimeFitPulse!=0   )     {delete mH1_TimeFitPulse;}
+
+  if( mH2_HeightvsSigma!=0  )     {delete mH2_HeightvsSigma;}
+  if( mH2_HeightvsSigmaTrig!=0 )  {delete mH2_HeightvsSigmaTrig;}
+  if( mH1_ChiNdf!=0         )     {delete mH1_ChiNdf;}
+  if( mH2_HeightvsChiNdf!=0 )     {delete mH2_HeightvsChiNdf;}
+  if( mH2_MeanvsChiNdf!=0   )     {delete mH2_MeanvsChiNdf;}
+  if( mH2_SigmavsChiNdf!=0  )     {delete mH2_SigmavsChiNdf;}
+
+  if( mH1_PeakTimingGaus!=0 )     {delete mH1_PeakTimingGaus;}
+  if( mH1_PeakTimingPuls!=0 )     {delete mH1_PeakTimingPuls;}
+  if( mH2_PeakTimingCompare!=0 )  {delete mH2_PeakTimingCompare;}
```

Comment:
I thought we already discussed this: there is no need to check for the pointer to be non null before deleting the object. Isn't nicer to have less code than more code?

```suggestion
  delete mH1_PeakTimingGaus;
  delete mH1_PeakTimingPuls;
  delete mH2_PeakTimingCompare;
```

Reply:
Done. Sorry I had forgotten about this file :)